components/sapi/sapi.h

```diff
@@ -0,0 +1,20 @@
+#ifndef DATADOG_PHP_SAPI_H
+#define DATADOG_PHP_SAPI_H
+
+#include "string_view/string_view.h"
+
+typedef enum {
+    DATADOG_PHP_SAPI_UNKNOWN = 0,
+    DATADOG_PHP_SAPI_APACHE2HANDLER,
+    DATADOG_PHP_SAPI_CGI_FCGI,
+    DATADOG_PHP_SAPI_CLI,
+    DATADOG_PHP_SAPI_CLI_SERVER,
+    DATADOG_PHP_SAPI_EMBED,
+    DATADOG_PHP_SAPI_LITESPEED,
+    DATADOG_PHP_SAPI_FPM_FCGI,
+    DATADOG_PHP_SAPI_PHPDBG,
+} datadog_php_sapi;
```

Comment:
Do you foresee any issues/things breaking with adding more SAPIs in the future? For example if we wanted to add the fuzzer SAPI or any of the pre-PHP 7 SAPIs?

Reply:
We just have to add them. Approving specific SAPIs seems better than rejecting specific ones as SAPIs seem to come and go (though mostly go lately).